enhancements/cluster-api/installing-cluster-api-components-in-ocp.md

```diff
@@ -0,0 +1,290 @@
+    - direct apply (through client-go) of the resulting CAPI providers manifests at runtime
+    - continuous tracking of the provider-list, the transport ConfigMap(s) and reconciliation of the applied CAPI manifests
+
+![runtime](./assets/proposed-runtime.png)
+
+### Workflow Description
+
+As such here is the new proposed workflow to develop and integrate a Cluster API provider into OpenShift:
+
+1. (To be done once) A Cluster API (CAPI) provider repository has to be created in the OpenShift org. The provider should be a syncable fork of an upstream CAPI provider and should adhere to the [`clusterctl` Provider Contract](https://cluster-api.sigs.k8s.io/clusterctl/provider-contract). The repository must then be configured to end up being build and included in the OpenShift payload.
+1. A Periodics Job on the Prow CI runs at regular intervals and launches the [Rebase Bot](https://github.com/openshift-eng/rebasebot). The bot performs a syncronization of the commits from a point release of the upstream repository corresponding to the fork by creating a rebasing the existing main branch and stacking the carried patches on top of the branch.
+1. Once the main branch has been rebased locally, the Rebase Bot invokes the relevant Lifecycle Hook (e.g. post-rebase) an executes the corresponding command defined for this stage, set to be the manifests generation tool (which resides in the [Cluster CAPI Operator](https://github.com/openshift/cluster-capi-operator) repository, and it is imported in the provider as a tool),
+which generates the manifests for the CAPI Provider (Deployments, RBACs, CRDs), it embeds them into the transport ConfigMaps, and dumps them to the `/openshift/manifests` folder in the Provider source tree. **TODO(@damdo): describe whether the CRDs/RBACs are transported (and how) or installed by the CVO**
+1. The Rebase Bot then proceeds creating a Pull Request (PR) containing the rebase changes to the main branch and the generated manifests. The PR then gets reviewed and merged into the main branch.
+1. Shortly after, the Provider's repository gets built by the ART builders and ends up being included into the payload. Also the transport ConfigMap(s) end up being embedded in the payload.
+1. Once the payload is installed and a new cluster is created the CVO kicks in and applies the manifests embedded introduced by the CAPI Provider repository and embedded in the payload. This results in the creation of the transport ConfigMap(s) in the APIServer. At this point the CVO also launches the [Cluster CAPI Operator](https://github.com/openshift/cluster-capi-operator).
+1. The Cluster CAPI Operator controllers start up. The _CAPI Installer controller_ loads the `capi-providers-list` ConfigMap from the APIServer (If none exists, a default one gets pre-created by the controller on the basis of what platform is detected in the Infrastructure Object's status).
```

Comment:
Why does it need this? Rather, should it not check the platform that it's running on via the `Infrastructure` and then use some well known label format to identify the correct configmaps to load resources from?

Reply:
Yes, what you describe is going to be the up to date step.
The one described there at the moment is out of date considering the most recent conversation we had on other comment threads here.

Reply:
This is now in sync with the latest discussions and the PoC.